eng/common/TestResources/remove-test-resources.yml

```diff
@@ -1,57 +1,27 @@
-# Removes resources from a cloud type specified by the variable (not parameter)
-# 'CloudType'. Use this as part of a matrix to remove resources from a
-# particular cloud instance. Normally we would use template variables instead of
-# parameters but matrix variables are not available during template expansion
-# so any benefits of parameters are lost.
-
 # Assumes steps in deploy-test-resources.yml was run previously. Requires
 # environment variable: AZURE_RESOURCEGROUP_NAME and Az PowerShell module
 
 parameters:
   ServiceDirectory: ''
+  SubscriptionConfiguration: sub-config-azure-cloud-test-resources
 
 steps:
-  - pwsh: >
-      eng/common/TestResources/Remove-TestResources.ps1
-      -ResourceGroupName "${env:AZURE_RESOURCEGROUP_NAME}"
-      -TenantId '$(aad-azure-sdk-test-tenant-id)'
-      -SubscriptionId '$(azure-subscription-id)'
-      -ProvisionerApplicationId '$(aad-azure-sdk-test-client-id)'
-      -ProvisionerApplicationSecret '$(aad-azure-sdk-test-client-secret)'
-      -ServiceDirectory '${{ parameters.ServiceDirectory }}'
-      -Environment 'AzureCloud'
-      -Force
-      -Verbose
-    displayName: Remove test resources (AzureCloud)
-    condition: and(ne(variables['AZURE_RESOURCEGROUP_NAME'], ''), eq(variables['CloudType'], 'AzureCloud'))
-    continueOnError: true
-
-  - pwsh: >
-      eng/common/TestResources/Remove-TestResources.ps1
-      -ResourceGroupName "${env:AZURE_RESOURCEGROUP_NAME}"
-      -TenantId '$(aad-azure-sdk-test-tenant-id-gov)'
-      -SubscriptionId '$(azure-subscription-id-gov)'
-      -ProvisionerApplicationId '$(aad-azure-sdk-test-client-id-gov)'
-      -ProvisionerApplicationSecret '$(aad-azure-sdk-test-client-secret-gov)'
-      -ServiceDirectory '${{ parameters.ServiceDirectory }}'
-      -Environment 'AzureUSGovernment'
-      -Force
-      -Verbose
-    displayName: Remove test resources (AzureUSGovernment)
-    condition: and(ne(variables['AZURE_RESOURCEGROUP_NAME'], ''), eq(variables['CloudType'], 'AzureUSGovernment'))
-    continueOnError: true
+  # On MacOS there are problems with access to the ~/.Azure folder. This
+  # workaround fixes the problem.
+  - bash: sudo rm -rf ~/.Azure
+    condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+    displayName: Clean out ~/.Azure
```

Comment:
This appears to be a known issue with AzureRM modules I'm not sure why we've suddenly begun encountering it here when we have never seen it before. 

When running `Connect-AzAccount` we see this message: 
```
WARNING: The provided service principal secret will be included in the 'AzureRmContext.json' file found in the user profile ( /Users/runner/.Azure ). Please ensure that this directory has appropriate protections.
```

The Mac job does fail in the absence of this step: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=383962&view=logs&j=e44bd064-d8d9-5feb-6343-09b8c49ab1a8&t=7b42e473-ec9c-5ec0-5373-db64d8009993

Reply:
Does it work if you pass `-Force`? That's the typical way in powershell to say, "yeah, I know - do it anyway". That's why I support `-Force` in our own scripts.

Reply:
The `-Force` does work. https://dev.azure.com/azure-sdk/internal/_build/results?buildId=388893&view=logs&j=e44bd064-d8d9-5feb-6343-09b8c49ab1a8&t=7b42e473-ec9c-5ec0-5373-db64d8009993